actors/market/src/lib.rs

```diff
@@ -370,40 +356,59 @@ impl Actor {
                 continue;
             }
 
-            // check VerifiedClient allowed cap and deduct PieceSize from cap
-            // drop deals with a DealSize that cannot be fully covered by VerifiedClient's available DataCap
-            if deal.proposal.verified_deal {
-                if let Err(e) = rt.send(
-                    &VERIFIED_REGISTRY_ACTOR_ADDR,
-                    crate::ext::verifreg::USE_BYTES_METHOD as u64,
-                    RawBytes::serialize(UseBytesParams {
-                        address: Address::new_id(client_id),
-                        deal_size: BigInt::from(deal.proposal.piece_size.0),
-                    })?,
-                    TokenAmount::zero(),
-                ) {
-                    info!("invalid deal {}: failed to acquire datacap exitcode: {}", di, e);
-                    continue;
+            // For verified deals, transfer datacap tokens from the client
+            // to the verified registry actor along with a specification for the allocation.
+            // Drop deal if the transfer fails.
+            // This could be done in a batch, but one-at-a-time allows dropping of only
+            // some deals if the client's balance is insufficient, rather than dropping them all.
```

Comment:
I am ok with this as it is since not batching is current behavior but dropping only some deals if the balance is insufficient rather than dropping all is definitely something that can be handled by the verifreg's receiver hook in the same way is is done here and that is probably the best thing to do.

Reply:
`validate_alloc_req` could have the same `drop if failed` semantics

Reply:
I agree it could, and perhaps worthwhile as a follow-up (likely for subsequent network version). The verifreg would have to send back unused datacap as change, but that should be possible.

Reply:
Kind of. The datacap token transfer will just fail if the balance is insufficient – it won't make it to verifreg. But here we could first check the balance/allowance and then only attempt an amount that will succeed.

Reply:
https://github.com/filecoin-project/builtin-actors/issues/662